.github/workflows/integration-test.yaml

```diff
@@ -62,6 +62,15 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y p7zip-full zlib1g-dev libssl-dev #needed by setup python
           pwsh #validate that powershell installed correctly
+      - uses: pnpm/action-setup@v3
+        if: ${{ inputs.runs-on == 'self-hosted' }}
+        with:
+          package_json_file: 'frontend/package.json'
+      - uses: actions/setup-node@v4
```

Comment:
The https://github.com/marketplace/actions/setup-pnpm docs show running setup-node before running the pnpm setup action. However, the setup-node docs say "package manager should be preinstalled" if you're using the `cache` input, so this should be okay if node comes preinstalled on the self-hosted runner.

Reply:
Solved: we had to run setup-node twice. Once before setting up pnpm (without the `cache` input param), and once afterwards with the `cache: pnpm` setting.